faculty_cli/cli.py

```diff
@@ -1559,8 +1570,31 @@ def publish():
 @click.argument("source_directory", default=os.getcwd(), required=False)
 def publish_new_template(template, source_directory):
     """Publish a new template from a directory to the knowledge centre."""
-    print(template)
-    print(source_directory)
+    template_client = faculty.client("template")
+    notification_client = faculty.client("notification")
+
+    user_id = _get_authenticated_user_id()
+    project_id = get_context().project_id
+
+    if not project_id:
+        _print_and_exit("Project ID must be set using FACULTY_PROJECT_ID.", 64)
```

Comment:
Do you think we should also mention that this command needs to be run on a Faculty server in this message?

Reply:
It's not technically required, but it's probably better for UX, so I'll add it.

Reply:
I'd probably omit the fact that it's read from `FACULTY_PROJECT_ID` completely, otherwise people might try to just set `FACULTY_PROJECT_ID` without really understanding that they need to run this command from inside the platform.